### ISSUE TITLE
DPM++ leaves residual noise in SDXL images, which is unexpected

### ISSUE THREAD

**Issue by nhnt11** [huggingface/diffusers#5689]
### Describe the bug

When using DPM++ with SDXL, there is residual noise in the result.

Upon investigation, my colleague @CodeCorrupt and I found that the final sigma was non-zero which we empirically deemed to be the culprit - i.e. when we set the final sigma to 0 artificially, the images were clean.

### Reproduction

```python3
import torch
from diffusers import StableDiffusionXLPipeline
from typing import cast
from diffusers import DPMSolverMultistepScheduler, StableDiffusionPipeline, 

sdxl_model = cast(StableDiffusionXLPipeline, StableDiffusionXLPipeline.from_pretrained(
    'stabilityai/stable-diffusion-xl-base-1.0',
    torch_dtype=torch.float16,
    use_safetensors=True,
    variant="fp16",
    revision="76d28af79639c28a79fa5c6c6468febd3490a37e",
)).to('cuda')
common_config = {'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear'}
scheduler =  DPMSolverMultistepScheduler(**common_config)

generator = torch.Generator(device='cuda')
sdxl_model.scheduler = scheduler
generator.manual_seed(96404382)
params = {
    'prompt': ['clorful abstract background, glowing gradient waves on black background'],
    'negative_prompt': [''],
    "num_inference_steps": 40,
    "guidance_scale": 8,
    "width": 1024,
    "height": 1024
}
sdxl_res = sdxl_model(**params, generator=generator, output_type='pil')
sdxl_img = sdxl_res.images[0]
display(sdxl_img)
```

Output with diffusers `main`: 
![image](https://github.com/huggingface/diffusers/assets/485488/136d7ee9-7864-4d22-8381-4d5d3ef80ee0)

Output with a hack to set the last sigma to 0:
![image](https://github.com/huggingface/diffusers/assets/485488/82bd3da4-1c48-45ba-b900-0fd3c15db2c9)

### System Info

- diffusers version: 0.22.1
- Platform: Linux-5.4.0-166-generic-x86_64-with-glibc2.31
- Python version: 3.11.5
- PyTorch version (GPU?): 2.1.0+cu121 (True)
- Huggingface_hub version: 0.17.1
- Transformers version: 4.34.0
- Accelerate version: 0.22.0
- xFormers version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help?

@yiyixuxu @patrickvonplaten

**Comment by nhnt11** [comment]
Here is the code where we manually set the last sigma to 0: https://github.com/playgroundai/diffusers/commit/3b4a2fd460bdf7b58e2277c663dbf62012a39df8

**Comment by bghira** [comment]
well that wouldn't be DPM++ then, now would it? :D it's a different scheduler you made. you probably want to rescale the betas.

**Comment by nhnt11** [comment]
@bghira Actually, the DPM++ implementation is already technically "not DPM++" by that definition if you use the `euler_at_final` flag. I think there are several approximations in place to increase numerical stability of these schedulers that admittedly compromise the theory.

Note that for many other samplers, diffusers explicitly appends a 0 sigma. My understanding is that this is necessary due to the way the diffusion loop is implemented: for every loop iteration, the unet pass happens before stepping the scheduler, and so the final iteration will step the scheduler before returning the result. If the scheduler adds noise in that final step, you get a noisy result.

Opinion: I think it's a valid tradeoff when implementing math in finite precision computing environments :)

**Comment by yiyixuxu** [comment]
as discussed here you can use `use_lu_lambdas`  and `euler_at_final` to avoid this issue
https://github.com/huggingface/diffusers/pull/5541

thanks :)

**Comment by nhnt11** [comment]
Hey @yiyixuxu, thanks for the pointer - but we're already using those. For clarity - we needed to add this hack to zero out the last sigma ON TOP of the `euler_at_final` option (or with Lu lambdas, or with Karras sigmas).

**Comment by spezialspezial** [comment]
It brings a bit of relief but using_lu_lambdas and euler_at_final does not solve this issue. Thanks to [nhnt11](https://github.com/nhnt11) for the investigation!

**Comment by mar-muel** [comment]
@nhnt11 Interesting! But is this issue maybe not related to poor performance of DPM++ for SDXL in general?

For example, in Automatic1111 (based on the k-diffusion repo) I'm seeing similar artefacting for the `DPM++ 2M` scheduler:
![00092-42](https://github.com/huggingface/diffusers/assets/19345805/d4720361-bc15-4c40-8183-d0144d08763c)
```
"colorful abstract background, glowing gradient waves on black background"
Steps: 20, Sampler: DPM++ 2M, CFG scale: 7.5, Seed: 42, Size: 1024x1024, Model hash: 31e35c80fc, Model: sd_xl_base_1.0, Version: v1.7.0
```

But then using `DPM++ 2M Karras` gives
![00093-42](https://github.com/huggingface/diffusers/assets/19345805/f221de38-942d-459c-9a8f-33292a54e814)
```
"colorful abstract background, glowing gradient waves on black background"
Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7.5, Seed: 42, Size: 1024x1024, Model hash: 31e35c80fc, Model: sd_xl_base_1.0, Version: v1.7.0
```

I agree with you that `use_lu_lambdas`/`euler_at_final` are more of an extension to DPM++ and not really a "fix" for it.

**Comment by patrickvonplaten** [comment]
This PR should have also improved the quality: https://github.com/huggingface/diffusers/pull/6477

**Comment by bghira** [comment]
it has not

**Comment by yiyixuxu** [comment]
@bghira 
let us know if you have any suggestions to further improve the DPM scheduler for SDXL
